### PR TITLE
feat(observability): telemetry counters for incentives and sqs

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Added Prometheus counters and logs for incentives and SQS flows which we recently spent considerable time debugging recently.

Additionally, fixed an old counter name that was causing pains for @alessandrolomanto 

Lastly, moved CL telemetry placeholders into a separate file for consistency

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/pool-incentives/keeper/keeper.go [hunk 1]
```diff
@@ -255,6 +256,8 @@ func (k Keeper) GetPoolIdFromGaugeId(ctx sdk.Context, gaugeId uint64, lockableDu
 	bz := store.Get(key)
 
 	if len(bz) == 0 {
+		telemetry.IncrCounter(1, types.NoPoolIdForGaugeTelemetryName)
+		ctx.Logger().Error(types.NoPoolIdForGaugeTelemetryName, "gaugeId", gaugeId, "duration", lockableDuration)
```
Comment (p0mvn) [inline x/pool-incentives/keeper/keeper.go:260]: @alessandrolomanto 

1. Let's make an alert for this counter please `pool_incentives_no_pool_id_for_gauge`. No dashboards needed

File: x/incentives/keeper/distribute.go [hunk 2]
```diff
@@ -44,7 +45,8 @@ func (k Keeper) AllocateAcrossGauges(ctx sdk.Context, activeGroups []types.Group
 	for _, group := range activeGroups {
 		err := k.syncGroupWeights(ctx, group)
 		if err != nil {
-			ctx.Logger().Error("error syncing group gauge weights, skipping", "group gauge id", group.GroupGaugeId, "error", err.Error())
+			telemetry.IncrCounter(1, types.SyncGroupGaugeFailureTelemetryName)
+			ctx.Logger().Error(types.SyncGroupGaugeFailureTelemetryName, "group gauge id", group.GroupGaugeId, "error", err.Error())
```
Comment (p0mvn) [inline x/incentives/keeper/distribute.go:49]: @alessandrolomanto 

2. Let's make an alert for this counter please `incentives_group_gauge_sync_failure`. No dashboards needed

File: ingest/sqs/pools/ingester/pool_ingester.go [hunk 3]
```diff
@@ -322,7 +322,7 @@ func (pi *poolIngester) convertPool(
 	defer func() {
 		r := recover()
 		if r != nil {
-			telemetry.IncrCounter(1, "sqs_ingest", "convert_pool", "panic", strconv.FormatUint(pool.GetId(), 10))
+			telemetry.IncrCounter(1, "sqs_ingest_convert_pool_panic", "pool_"+strconv.FormatUint(pool.GetId(), 10))
```
Comment (p0mvn) [inline ingest/sqs/pools/ingester/pool_ingester.go:325]: @alessandrolomanto 

3. Alert and no dashboard please 🙏 

Name should be `sqs_ingest_convert_pool_panic`

File: x/incentives/keeper/distribute.go (hunk 2 above)
Comment (alessandrolomanto) [reply x/incentives/keeper/distribute.go:49]: Could you pls edit the string `group gauge id` to `group_gauge_id`?

File: x/pool-incentives/keeper/keeper.go (hunk 1 above)
Comment (alessandrolomanto) [reply x/pool-incentives/keeper/keeper.go:260]: What values can the counter take: 0 or 1? Should the alert be triggered when it differs from 0?
For this case I prefer to create alert based on logs

File: x/incentives/keeper/distribute.go (hunk 2 above)
Comment (alessandrolomanto) [reply x/incentives/keeper/distribute.go:49]: What values can the counter take: 0 or 1? Should the alert be triggered when it differs from 0?
For this case I prefer to create alert based on logs.
Eg: Error syncing group gauge with id: < group_gauge_id>

File: ingest/sqs/pools/ingester/pool_ingester.go (hunk 3 above)
Comment (alessandrolomanto) [reply ingest/sqs/pools/ingester/pool_ingester.go:325]: What values can the counter take: 0 or 1? Should the alert be triggered when it differs from 0?

Comment (p0mvn) [reply ingest/sqs/pools/ingester/pool_ingester.go:325]: If it increases, we should get alerted

File: x/incentives/keeper/distribute.go (hunk 2 above)
Comment (p0mvn) [inline x/incentives/keeper/distribute.go:49]:
```suggestion
			ctx.Logger().Error(types.SyncGroupGaugeFailureTelemetryName, "group_gauge_id", group.GroupGaugeId, "error", err.Error())
```

File: x/pool-incentives/keeper/keeper.go (hunk 1 above)
Comment (p0mvn) [reply x/pool-incentives/keeper/keeper.go:260]: Based on logs works for me!

File: x/incentives/keeper/distribute.go (hunk 2 above)
Comment (p0mvn) [reply x/incentives/keeper/distribute.go]: If it increases, we should get alerted

**Review by czarcas7ic — APPROVED** [review]

**Pull request merged**